internal/types/kruizePayload/common.go

```diff
@@ -91,103 +91,160 @@ func convertMetricToString(data interface{}) string {
-				Results: result{
-					Aggregation_info: aggregation_info{
-						Sum:    convertMetricToString(c["memory_request_container_sum_SUM"]),
-						Avg:    convertMetricToString(c["memory_request_container_avg_MEAN"]),
-						Format: "MiB",
-					},
+		})
+	}
+
+	// memoryRequest
+	sum = convertMetricToString(c["memory_request_container_sum_SUM"])
+	avg = convertMetricToString(c["memory_request_container_avg_MEAN"])
+	if sum != "" && avg != "" {
+		metrics = append(metrics, metric{
+			Name: "memoryRequest",
+			Results: result{
+				Aggregation_info: aggregation_info{
+					Sum:    sum,
+					Avg:    avg,
+					Format: "MiB",
 				},
 			},
-			{
-				Name: "memoryLimit",
-				Results: result{
-					Aggregation_info: aggregation_info{
-						Sum:    convertMetricToString(c["memory_limit_container_sum_SUM"]),
-						Avg:    convertMetricToString(c["memory_limit_container_avg_MEAN"]),
-						Format: "MiB",
-					},
+		})
+	}
+
+	// memoryLimit
+	sum = convertMetricToString(c["memory_limit_container_sum_SUM"])
+	avg = convertMetricToString(c["memory_limit_container_avg_MEAN"])
+	if sum != "" && avg != "" {
+		metrics = append(metrics, metric{
+			Name: "memoryLimit",
+			Results: result{
+				Aggregation_info: aggregation_info{
+					Sum:    sum,
+					Avg:    avg,
+					Format: "MiB",
 				},
 			},
-			{
-				Name: "memoryUsage",
-				Results: result{
-					Aggregation_info: aggregation_info{
-						Min:    convertMetricToString(c["memory_usage_container_min_MIN"]),
-						Max:    convertMetricToString(c["memory_usage_container_max_MAX"]),
-						Sum:    convertMetricToString(c["memory_usage_container_sum_SUM"]),
-						Avg:    convertMetricToString(c["memory_usage_container_avg_MEAN"]),
-						Format: "MiB",
-					},
+		})
+	}
+
+	// memoryUsage
+	sum = convertMetricToString(c["memory_usage_container_sum_SUM"])
+	avg = convertMetricToString(c["memory_usage_container_avg_MEAN"])
+	min = convertMetricToString(c["memory_usage_container_min_MIN"])
+	max = convertMetricToString(c["memory_usage_container_max_MAX"])
+	if sum != "" && avg != "" {
+		metrics = append(metrics, metric{
+			Name: "memoryUsage",
+			Results: result{
+				Aggregation_info: aggregation_info{
+					Min:    min,
+					Max:    max,
+					Sum:    sum,
+					Avg:    avg,
+					Format: "MiB",
 				},
 			},
-			{
-				Name: "memoryRSS",
-				Results: result{
-					Aggregation_info: aggregation_info{
-						Min:    convertMetricToString(c["memory_rss_usage_container_min_MIN"]),
-						Max:    convertMetricToString(c["memory_rss_usage_container_max_MAX"]),
-						Sum:    convertMetricToString(c["memory_rss_usage_container_sum_SUM"]),
-						Avg:    convertMetricToString(c["memory_rss_usage_container_avg_MEAN"]),
-						Format: "MiB",
-					},
+		})
+	}
+
+	// memoryRSS
+	sum = convertMetricToString(c["memory_rss_usage_container_sum_SUM"])
+	avg = convertMetricToString(c["memory_rss_usage_container_avg_MEAN"])
+	max = convertMetricToString(c["memory_rss_usage_container_min_MIN"])
```

Comment:
Please correct me if I'm wrong. I would expect `max` to be mapped with something like `memory_rss_usage_container_max_MAX` as I found out that `memory_rss_usage_container_min_MIN` is already being used for `min`, Also in the earlier code the `max` was set to the return value of this `convertMetricToString(c["memory_rss_usage_container_max_MAX"])`

Reply:
Oh, that's correct. Thanks for pointing it out. I will make the changes.